src/lib/y2network/config.rb

```diff
@@ -129,6 +129,17 @@ def ==(other)
         routing == other.routing && dns == other.dns
     end
 
+    # Renames a given interface and the associated connections
+    #
+    # @param old_name  [String] Old interface's name
+    # @param new_name  [String] New interface's name
+    # @param mechanism [Symbol] Property to base the rename on (:mac or :bus_id)
+    def rename_interface(old_name, new_name, mechanism)
+      interface = interfaces.by_name(old_name)
+      interface.rename(new_name, mechanism)
+      connections.by_interface(old_name).each { |c| c.interface = new_name }
```

Comment:
btw we probably need to update also hostname  tab where it can link to interface. Or does we use object there?

Reply:
I will check.

Reply:
I've created [a PBI](https://trello.com/c/2ZfEE3yB/3456-fix-dhcpsethostname-handling) a card because dhcp_hostname is not handled correctly.